### ISSUE TITLE
Set-up OAuth Github client to authenticate user

### ISSUE THREAD

**Issue by abiduzz420** [aravindballa/node-gister#5]
I would like to take up this task. Please assign it to me

**Comment by abiduzz420** [comment]
@aravindballa You can follow this issue as well ![here](https://github.com/Demfier/GitPub/issues/1)

**Comment by abiduzz420** [comment]
How would you like the user to interact with `node-gister`:
1. Should the user manually input his username and password everytime he wants to make a `POST /gists` request ? In that case we will have to expose `clientID` and `clientSecret` of the application  OR
2. Do you want the user to generate his OAuthToken from Github UI with `gists` scope, store this info in `.env` file along with username and then he will never be asked to enter anything. he can directly make a `POST` request. Not sure how this would work with npm
Which one would @aravindballa you prefer ?

**Comment by aravindballa** [comment]
First option.

We can have an instance deployed with `clientID` and `clientSecret` in that but will not have it on Github. Let's store it in `.env` file and only have a dummy `.env.example` file on Github so that if any wants to build it locally, they can do it with replacing their tokens in that file.

**Comment by abiduzz420** [comment]
`clientId` and `clientSecret` is specific to a Github App. So we cannot let user create his own `clientId` and `clientSecret`.

**Comment by aravindballa** [comment]
Yeah. I have one more question. If we generate `clientId` and `clientSecret` can anyone else login using that? No, right?

**Comment by abiduzz420** [comment]
Yeah they will be able to login provided they have their own access tokens.

**Comment by abiduzz420** [comment]
I would prefer Option 2:

1. User install `node-gister` through `npm install -g node-gister`
2. When he executes `node-gister` we will prompt him to Step 3 for the first time. 
3. Ask the developer to create `.env` file and store his username and access token. We shall include the documentation of generating personal access tokens.
4. After this, he can always make `POST /gists` without any issues.

**Comment by abiduzz420** [comment]
What I'm unsure of is, in which directory should the user create the `.env` file after installing the package globally ? Any idea ?

**Comment by aravindballa** [comment]
I am just wondering how do some tools take in the username and password from the commandline itself and do things like git push and all.

**Comment by abiduzz420** [comment]
We can do the same provided we expose the `clientId` and `clientSecret`. I am not sure if it is a good practice to expose such data.

Meanwhile I got to know where the global npm packages will be installed using the command `npm root -g`
Going into the `node-gister` directory, user will be able to create `.env` file.

**Comment by aravindballa** [comment]
Good to know the progress. 

Even if we expose them, can anyone login using it?

**Comment by abiduzz420** [comment]
If we expose that data then any user can log in.
Have a look at this : https://github.com/coderplex/tweet-issues/blob/master/README.md

**Comment by aravindballa** [comment]
Regarding https://github.com/coderplex/tweet-issues/blob/master/README.md

I don't think it's for any user with just one clientId, more of like a clientId for anyone who wants to deploy a bot.
